src/dataframe/dataframe.jl

```diff
@@ -1317,3 +1317,77 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+function _expandhelper(col, vect, stride, offset)
+    for i in offset:stride:length(vect)
+        push!(col, vect[i])
+    end
+end
+
+function expand(df::AbstractDataFrame, indexcols)
```

Comment:
@nalimilan - I am not sure here if `expand` and `complete` are the best names. What do you think. Maybe better keep the `df` suffix?

Reply:
We don't use the "df" suffix anywhere (except in `stackdf` that we want to remove), and that's not really Julian since that hardcodes the type in the function name, so I prefer `expand` (or another name). We could define it in DataAPI at some point. Regarding `complete`, I suggested above that it could be replaced with a keyword argument to `expand` to limit the number of exported functions. What do you think?

Reply:
I am OK with your proposal (and we can move `expand` to DataAPI.jl later when it is needed).